### DISCUSSION TITLE
Implement timing based colour coded hit circles to osu! (STD)

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#15220] [Ideas]
Mania semi recently had the ability to colour notes depending on what time they occupy in the time signature, a feature STD currently lacks. having 2/4 beats 1/4 1/3 etc... coloured differently would aid the readability of rythmically complex maps in addition to slightly increasing the variety of ways one can "read" a beatmap.

I think the idea is clear but if it isnt ill happily draw something up in figma and link it

**Comment by smoogipoo** [comment]
Maybe can make use of the existing setting for osu!mania (`ManiaRulesetSetting.TimingBasedNoteColouring`). Should be a relatively simple implementation.

**Reply by mk56-spn** [reply]
i was considering that, ill try and see if it can be done later, this discussion was mainly to know if its something you guys would allow, which it seems to be

**Reply by peppy** [reply]
Definitely. It should replace existing combo colours in osu! ruleset.

**Reply by mk56-spn** [reply]
> Definitely. It should replace existing combo colours in osu! ruleset.

sounds good, any ideas as to how this would interact with follow points? considering they are currently tied to new combos ?

**Reply by smoogipoo** [reply]
Imo followpoints shouldn't change, and I don't think it's worth discussing until this is actually implemented. Following the implementation of osu!mania as I proposed will result in followpoints not being affected.

**Reply by integerrr** [reply]
> 
> 
> Definitely. It should replace existing combo colours in osu! ruleset.

The only thing I'd say is watch out for the mapping community they might start yelling if it's been completely replaced

**Reply by mk56-spn** [reply]
> > Definitely. It should replace existing combo colours in osu! ruleset.
> 
> The only thing I'd say is watch out for the mapping community they might start yelling if it's been completely replaced

what im going to attempt is a mere toggle like mania has, none of the combo stuff mappers use will be touched

**Comment by mk56-spn** [comment]
https://www.youtube.com/watch?v=WuA4BjaPoq4&ab_channel=Daysofthunder

**Reply by mk56-spn** [reply]
Here is the implementation as I currently have it, any issues?

**Reply by bdach** [reply]
doesn't look half bad on this particular skin, but how about others?

**Reply by mk56-spn** [reply]
same issue the mania implementation has, it looks goddamn awful on skins with colours baked in ( anything non white / greyscale ), but as far as im concerned thats unavoidable with the current skin system, it also happens with new combo colours from maps

**Reply by peppy** [reply]
I'm confused. This looks to just be recolouring the whole hitobject, which won't work with combo colours. Sounds like you aren't applying to the accent/combo colour? If so, I highly doubt this is what we want.

**Reply by mk56-spn** [reply]
> I'm confused. This looks to just be recolouring the whole hitobject, which won't work with combo colours. Sounds like you aren't applying to the accent/combo colour? If so, I highly doubt this is what we want.

This whole time I've been working undert he understanding that it was an alternative to combo colours,not complimentary. Having both colours on the circle would create a hard to read mess since this is substituting the reading part involved with the combo colours.

**Reply by bdach** [reply]
No, I believe the proposal above was that the timing-based colour should *replace* standard combo colours.

**Reply by mk56-spn** [reply]
can I PR to get a better understanding of where the issues are? , even if it doesnt get accepted  ( Which i hope it will since theres been overwhelming support everywhere ive shown it) it would make me grasping the issues easier

**Reply by mk56-spn** [reply]
https://github.com/mk56-spn/osu i fixed it to work on coloured skins and fixed the issue peppy mentioned about it colouring the whole hitcircle

https://www.youtube.com/watch?v=30-IITUrvyE&ab_channel=Daysofthunder
